src/utils/actions.js

```diff
@@ -48,65 +49,72 @@ const _paginatedNetworkFetch = async (url, baseUrl, req, parse) => {
-        }),
-        headers: {
-            ...(token ? {"Authorization": `Bearer ${token}`} : {}),
-            ...(req?.headers ?? {}),
-        },
-    };
-
-    let {parse} = fnResult;
-    if (!parse) parse = r => r.json();
-
-    dispatch({type: types.REQUEST, ...params});
-    try {
-        const data = await (paginated ? _paginatedNetworkFetch : _unpaginatedNetworkFetch)(
-            url, baseUrl, finalReq, parse);
-        dispatch({
-            type: types.RECEIVE,
-            ...params,
-            ...(data === null ? {} : {data}),
-            receivedAt: Date.now(),
-        });
-        if (onSuccess) await onSuccess(data);
-    } catch (e) {
-        if (err) {
-            console.error(e, err);
-            message.error(err);
-        }
-        dispatch({type: types.ERROR, ...params, caughtError: e});
-        if (onError) await onError(e);
-    }
-    dispatch({type: types.FINISH, ...params});
-};
+const _networkAction =
+    (fn, ...args) =>
+        async (dispatch, getState) => {
+            let fnResult = await fn(...args);
+            if (typeof fnResult === "function") {
+            // Needs dispatch / getState, resolve those.
+                fnResult = await fnResult(dispatch, getState);
+            }
+
+            const { types, params, url, baseUrl, req, err, onSuccess, onError, paginated } = fnResult;
+
+        // Only include access token when we are making a request to this Bento node or the IdP!
+        // Otherwise, we could leak it to external sites.
+
+            const token =
+            url.startsWith("/") ||
+            (BENTO_URL && url.startsWith(BENTO_URL)) ||
+            (BENTO_PUBLIC_URL && url.startsWith(BENTO_PUBLIC_URL)) ||
+            (IDP_BASE_URL && url.startsWith(IDP_BASE_URL))
+                ? getState().auth.accessToken
+                : null;
+
+            const finalReq = {
+                ...(req ?? {
+                    method: "GET", // Default request method
+                }),
+                headers: {
+                    ...(token ? { Authorization: `Bearer ${token}` } : {}),
+                    ...(req?.headers ?? {}),
+                },
+            };
+
+            let { parse } = fnResult;
+            if (!parse) parse = (r) => r.json();
+
+            dispatch({ type: types.REQUEST, ...params });
+            try {
+                const data = await (paginated ? _paginatedNetworkFetch : _unpaginatedNetworkFetch)(
+                    url,
+                    baseUrl,
+                    finalReq,
+                    parse,
+                );
+                dispatch({
+                    type: types.RECEIVE,
+                    ...params,
+                    ...(data === null ? {} : { data }),
+                    receivedAt: Date.now(),
+                });
+                if (onSuccess) await onSuccess(data);
+            } catch (e) {
+                const errorMsg = e.message || err;
```

Comment:
the `err` parameter is used to add nice errors in actions, and this is confusing behaviour if it gets overwritten in some cases - maybe we should remove it completely, or use it as a prefix like `${err}: ${e.message}` if `err` is set